src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -0,0 +1,111 @@
+package frc.robot.subsystems;
+
+import frc.robot.SwerveModule;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+
+import com.ctre.phoenix.sensors.Pigeon2;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Swerve extends SubsystemBase {
+    public SwerveDriveOdometry swerveOdometry;
+    public SwerveModule[] mSwerveMods;
+    public Pigeon2 gyro;
+
+    public Swerve() {
+        gyro = new Pigeon2(RobotMap.Pigeon.PIGEON_ID);
```

Comment:
Does anyone else need the Pigeon, or is it really just this drive subsystem? If others need it, we'll want to factor it out into its own subsystem. If you know for sure that someone else will use it, rather than change this now, open an issue to get someone to factor it out. That'd be a nice, easy issue for anyone.

Reply:
Only other commands need it, not other subsystems. I am going to leave it here for now, but if that changes I will add that issue. Thanks!

Reply:
If there are times outside of the drive base that need the Pigeon/Gyro value, they should access it through the `Swerve.getPose()`, therefore I would also think about making `Swerve.getYaw()` private as that may be offset-ed through April tags, resets, etc.